basic csv importer

willdoran opened the pull request:
@jasonmule I've created the outline for the basic csv importer with tests :) Will finish it up tonight and we can discuss integrating tomorrow.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/ushahidi/platform-client/52)
<!-- Reviewable:end -->

rjmackay commented:
Haven't run the code yet.. but mostly looks pretty good. Few clean ups in my comments though.

---

Reviewed 13 of 13 files at r1.
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[app/setting/controllers/setting-data-import-controller.js, line 15 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TozGXlJN572XyPzUd:-K7TozGXlJN572XyPzUe:-841147191) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/controllers/setting-data-import-controller.js#L15)):*
Any reason this is inside the callback? seems unrelated

---

*[app/setting/controllers/setting-data-mapper-controller.js, line 32 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tpj-7dDn72XY12rL7:-K7Tpj-8OLDrcGf4YXLY:-941131736) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/controllers/setting-data-mapper-controller.js#L32)):*
Could this live in a service? what does it depend on?

Somone at a meetup recently suggested refactoring by move **every** function to its own file + module.. overkill but the general idea sorta works, always ask: could this function be completely independent?

---

*[app/setting/controllers/setting-data-mapper-controller.js, line 48 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TqQvfcCpxAU_BvCqx:-K7TqQvfcCpxAU_BvCqy:1443535935) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/controllers/setting-data-mapper-controller.js#L48)):*
since this loads based on route params you could move the loading into the resolve argument of the route.. that way it will all be loaded before the controller happens.

---

*[app/setting/controllers/setting-data-mapper-controller.js, line 72 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tqb0ubsarcqjexy3f:-K7Tqb0ubsarcqjexy3g:953582659) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/controllers/setting-data-mapper-controller.js#L72)):*
Any reason most of the logic in the controller couldn't go into a directive?

---

*[app/setting/directives/setting-data-import-directive.js, line 20 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tr8I-gX5qdAuDBd63:-K7Tr8I0NsJmzK3prNOJ:-252834176) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-import-directive.js#L20)):*
when does this condition happen? Personal preference, but it might be cleaner to pull the check out at the top ie.
```
if ($scope.fileContainer.file) {
showerror();
return;
}

// other code (not wrapped in an else anymore)
```

---

*[app/setting/directives/setting-data-import-directive.js, line 36 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tr1p8blPoqfnXBzuj:-K7Tr1p8blPoqfnXBzuk:1003558900) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-import-directive.js#L36)):*
fata_import -> data_import

---

*[app/setting/directives/setting-data-mapper-directive.js, line 19 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TrnqhjKExmrcgCDnS:-K7TrnqhjKExmrcgCDnT:1055325622) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-mapper-directive.js#L19)):*
is this a common pattern? ie. show notification + navigate
If you notice it turning up often we should make a generic handler

---

*[app/setting/directives/setting-data-mapper-directive.js, line 28 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TsN2kiFi_fv_n29nL:-K7TsN2kiFi_fv_n29nM:-414551610) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-mapper-directive.js#L28)):*
Do we *need* to do this here? could we just be using 'key' in the forms value to start with?

---

*[app/setting/directives/setting-data-mapper-directive.js, line 32 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TsbTr0rUutmfV93_m:-K7TsbTr0rUutmfV93_n:26150968) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-mapper-directive.js#L32)):*
needs a comment

---

*[app/setting/directives/setting-data-mapper-directive.js, line 36 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tsd6FzEfkoN8J252N:-K7Tsd6FzEfkoN8J252O:26150968) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-mapper-directive.js#L36)):*
needs a comment

---

*[server/www/templates/settings/data-import/data-import.html, line 31 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TtSOGJ1sHhA3PQgQq:-K7TtSOGJ1sHhA3PQgQr:-153560388) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/server/www/templates/settings/data-import/data-import.html#L31)):*
Best practice would be to move the ng-click to ng-submit on the `form` tag .. 

---

*[test/e2e/settings/csv-basic-spec.js, line 2 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TtgLITqfhQKXDDnoI:-K7TtgLJ7mqeDkCQVzJx:-462189928) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/test/e2e/settings/csv-basic-spec.js#L2)):*
commented out?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-:-K7Ttl0i7jGhLF1XCiRW:529522072)*
<!-- Sent from Reviewable.io -->

willdoran commented:
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[app/setting/controllers/setting-data-import-controller.js, line 15 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TozGXlJN572XyPzUd:-K7TxjWgYzyMfswrzEHH:1908879751) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/controllers/setting-data-import-controller.js#L15)):*
Nope, moved it.

---

*[app/setting/controllers/setting-data-mapper-controller.js, line 32 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tpj-7dDn72XY12rL7:-K7Txnx1womm8jxRFGEK:-512921014) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/controllers/setting-data-mapper-controller.js#L32)):*
Yeah, that's interesting. I think it would be a util for the data-import. It seems more util than service. So I've moved to it to util.

---

*[app/setting/controllers/setting-data-mapper-controller.js, line 48 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TqQvfcCpxAU_BvCqx:-K7TyPag94JE7-xs0rsU:-949712065) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/controllers/setting-data-mapper-controller.js#L48)):*
Interesting, we should do that for all controllers where this happens. 

---

*[app/setting/controllers/setting-data-mapper-controller.js, line 72 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tqb0ubsarcqjexy3f:-K7TyvW3gwGPhFy28_fM:-2059378080) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/controllers/setting-data-mapper-controller.js#L72)):*
None, with the two above gone.

---

*[app/setting/directives/setting-data-import-directive.js, line 20 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tr8I-gX5qdAuDBd63:-K7X7wmIPFXzbgyNW4LN:507749641) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-import-directive.js#L20)):*
I like it!

---

*[app/setting/directives/setting-data-import-directive.js, line 36 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tr1p8blPoqfnXBzuj:-K7X80zkPta3BNNiZGLi:677519329) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-import-directive.js#L36)):*
d'oh

---

*[app/setting/directives/setting-data-mapper-directive.js, line 19 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TrnqhjKExmrcgCDnS:-K7X87ojnb-c9wV-VpZv:1075823434) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-mapper-directive.js#L19)):*
Yeah, it is, I was thinking of extending the notification handler to include a post notify action option. This would probably mostly be navigate. 

I think I mentioned in the response to your email that flow is something I'd like to discuss when everyone is together. As it's slightly inconsistent across different saves.

---

*[app/setting/directives/setting-data-mapper-directive.js, line 28 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TsN2kiFi_fv_n29nL:-K7XDCqN2LLOJgfv65lV:-1654536789) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-mapper-directive.js#L28)):*
Yeah, we certainly could. In fact it's done!

---

*[app/setting/directives/setting-data-mapper-directive.js, line 32 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TsbTr0rUutmfV93_m:-K7XKQnw9-GRUVI2l6Wj:676635175) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-mapper-directive.js#L32)):*
Done

---

*[app/setting/directives/setting-data-mapper-directive.js, line 36 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7Tsd6FzEfkoN8J252N:-K7XKS2cJLM_32e6FFz4:676635175) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/app/setting/directives/setting-data-mapper-directive.js#L36)):*
Done

---

*[server/www/templates/settings/data-import/data-import.html, line 31 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TtSOGJ1sHhA3PQgQq:-K7XKoj7pYnfKtRUir9h:2030421964) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/server/www/templates/settings/data-import/data-import.html#L31)):*
Groovy I'll start to do that everywhere.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/ushahidi/platform-client/52)*
<!-- Sent from Reviewable.io -->

willdoran commented:
Review status: 8 of 14 files reviewed at latest revision, 12 unresolved discussions.

---

*[test/e2e/settings/csv-basic-spec.js, line 2 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/52#-K7TtgLITqfhQKXDDnoI:-K7XbBsWweYiTfxF3IE4:-2138161652) ([raw file](https://github.com/ushahidi/platform-client/blob/efde2136a3f4efac79344e57fc557f4ca8d07ded/test/e2e/settings/csv-basic-spec.js#L2)):*
Yeah, I've intentionally left this out because I wanted to discuss it at our next group meeting. The e2e tests are not...fast. 

I've tried to reduce them as far as possible but there is a lot of functions that need to be tested. Before I go further with it, I wanted us to all input on how we feel about the e2e tests and whether I could be handling them better; structurally or using a different test framework.

At the moment, the tests take a decent amount of time to run.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/ushahidi/platform-client/52)*
<!-- Sent from Reviewable.io -->
